### PR TITLE
[WIP] transfer methods to TileMatrixSet models

### PR DESCRIPTION
**Pull request by vincentsarago**

### CODE REVIEW

File: morecantile/models.py [hunk 1]
```diff
@@ -45,3 +62,272 @@ class TileMatrixSet(BaseModel):
...
+            1.0 if self.crs.linear_units == "metre" else 2 * math.pi * 6378137 / 360.0
+        )
+
+    @classmethod
+    def load(cls, name: str):
+        """Load default TileMatrixSet."""
+        try:
+            return cls.parse_file(os.path.join(data_dir, f"{name}.json"))
+        except FileNotFoundError:
+            raise NotAValidName(f"'{name}' is not a valid default TileMatrixSet.")
```
Comment (vincentsarago) [inline morecantile/models.py:82]: simple method to load a default TileMatrixSet

```python
import morecantile

tms = morecantile.TileMatrixSet.load("WebMercatorQuad")
tms.xy_bounds(morecantile.Tile(10, 10, 4))
```

File: morecantile/models.py [hunk 2]
```diff
@@ -45,3 +62,272 @@ class TileMatrixSet(BaseModel):
...
+        matrix_scale: list
+            Tiling schema coalescence coefficient (default: [1, 1] for EPSG:3857).
+            Should be set to [2, 1] for EPSG:4326.
+            see: http://docs.opengeospatial.org/is/17-083r2/17-083r2.html#14
+
+        Returns:
+        --------
+        TileMatrixSet
+        """
+        raise NotImplementedError
```
Comment (vincentsarago) [inline morecantile/models.py:113]: Placeholder right now

File: morecantile/models.py [hunk 3]
```diff
@@ -45,3 +93,278 @@ class TileMatrixSet(BaseModel):
...
+        Tile
+
+        """
+        matrix = self.matrix(zoom)
+
+        res = self._resolution(matrix)
+        xtile = int(
+            math.floor(
+                (xcoord - matrix.topLeftCorner[0])
+                / float(res * matrix.tileWidth * (matrix.matrixWidth / (2 ** zoom)))
```
Comment (vincentsarago) [inline morecantile/models.py:206]: need to validate `matrix.matrixWidth / (2 ** zoom)`

File: morecantile/models.py [hunk 4]
```diff
@@ -45,3 +93,278 @@ class TileMatrixSet(BaseModel):
...
+            see: http://docs.opengeospatial.org/is/17-083r2/17-083r2.html#14
+
+        Returns:
+        --------
+        TileMatrixSet
+
+        """
+        raise NotImplementedError
+
+    def matrix(self, zoom: int) -> TileMatrix:
```
Comment (geospatial-jeff) [inline morecantile/models.py:154]: Maybe use filter:

```python
        try:
            return list(filter(lambda m: m.identifier == str(zoom), self.tileMatrix))[0]
        except IndexError:
            raise Exception(f"TileMatrix not found for level: {zoom}")
```

File: morecantile/models.py [hunk 5]
```diff
@@ -45,3 +93,278 @@ class TileMatrixSet(BaseModel):
...
+        """Return the TileMatrix for a specific zoom."""
+        zoom_str = str(zoom)
+        for matrix in self.tileMatrix:
+            if matrix.identifier == zoom_str:
+                return matrix
+
+        raise Exception(f"TileMatrix not found for level: {zoom}")
+
+    def _resolution(self, matrix: TileMatrix) -> float:
```
Comment (geospatial-jeff) [inline morecantile/models.py:163]: From the docstring I would think this takes input of `zoom`

```python
return self.matrix(zoom) * 0.28e-3 / self.meters_per_unit
```

Comment (vincentsarago) [reply morecantile/models.py:161]: I edited the Docstring to 
> Tile resolution for a TileMatrix.

We could use `zoom` but then we will have to find the coresponding tileMatrix which is already done in most function calling `_resolution`

**Pull request merged**